app/src/main/java/com/nio/TcpDataFlowExample.java

```diff
@@ -1,117 +1,45 @@
 package com.nio;
 
-import com.google.common.collect.ArrayListMultimap;
-import com.google.common.collect.Multimap;
+import com.nio.tcp.entities.Port;
+import com.nio.tcp.entities.TcpConnectionData;
+import com.nio.tcp.entities.TcpDataFlowConnector;
 
-import java.net.InetSocketAddress;
-import java.nio.ByteBuffer;
+import java.io.IOException;
 import java.nio.channels.SelectionKey;
-import java.nio.channels.Selector;
-import java.nio.channels.ServerSocketChannel;
-import java.nio.channels.SocketChannel;
-import java.nio.charset.StandardCharsets;
-import java.util.Arrays;
-import java.util.Iterator;
 
 public final class TcpDataFlowExample {
```

Comment:
There is no way to use the service as API. The main method is blocked in the main thread.